### PR TITLE
Create multiple answer question

### PR DESCRIPTION
**Pull request by IlyaPrikhach**
Сделал вью для добавления вопросов в бд, и вью для редактирования вопроса из бд. Также контроллеры к ним. Использовал 2 метода отправки данных: form и ajax. Добавил метод MethodForAdd() в репозиторий и метод updateQuestion() в MultipleAnswerQuestion()

### CODE REVIEW

File: src/Teams/Controllers/MultipleAnswerQuestionController.cs [hunk 1]
```diff
@@ -37,5 +37,59 @@ public IActionResult MultipleAnswerQuestionForm(string id, int[] answers)
...
+        public IActionResult AddMultipleAnswerQuestion()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public IActionResult AddMultipleAnswerQuestion([FromBody] MAQDTOModel fromAjax)
+        {
+            List<MultipleAnswerQuestionOption> allAnswers = new List<MultipleAnswerQuestionOption>();
+            for (int i = 0; i < fromAjax.jTextList.Length; i++)
```
Comment (viktartse) [inline src/Teams/Controllers/MultipleAnswerQuestionController.cs:60]: Usually linq is easier to read. Could you try to replace the "for" loop with a linq and lambda expressions?

File: src/Teams/Controllers/MultipleAnswerQuestionController.cs [hunk 2]
```diff
@@ -37,5 +37,59 @@ public IActionResult MultipleAnswerQuestionForm(string id, int[] answers)
             };
             return View(question);
         }
+
+        public IActionResult EditMultipleAnswerQuestion(Guid id)
+        {
+            var question = new MultipleAnswerQuestionViewModel()
+            {
+                SourceQuestion = questionRepository.PickById(id),
+                // ChosenOptions = answers
```
Comment (viktartse) [inline src/Teams/Controllers/MultipleAnswerQuestionController.cs:46]: Do we need this comment?

File: src/Teams/Controllers/MultipleAnswerQuestionController.cs [hunk 3]
```diff
@@ -37,5 +37,59 @@ public IActionResult MultipleAnswerQuestionForm(string id, int[] answers)
...
+            questionRepository.MethodForAdd(new MultipleAnswerQuestion(fromAjax.questionText, allAnswers));
+
+            return View(); 
+            
+        }
+        [HttpPost]
+        public IActionResult EditMultipleAnswerQuestion( int[] editAnswers, string[] answersText, string questionText, Guid Id)
+        {
+            List<MultipleAnswerQuestionOption> allAnswers = new List<MultipleAnswerQuestionOption>();
+            int g = 0;
```
Comment (viktartse) [inline src/Teams/Controllers/MultipleAnswerQuestionController.cs:73]: What is the meaning of the "g" variable. Could you use more meaningful names for variables?

File: src/Teams/Domain/MultipleAnswerQuestion.cs [hunk 4]
```diff
@@ -32,5 +32,11 @@ public MultipleAnswerQuestionOption[] GetRightAnswers()
         {
             return Answers.Where(a => a.IsRight).ToArray();
         }
+
+        public void UpdateQuestion(string questionText, List<MultipleAnswerQuestionOption> Answers)
```
Comment (viktartse) [inline src/Teams/Domain/MultipleAnswerQuestion.cs:36]: Answers -> answers

File: src/Teams/Controllers/MultipleAnswerQuestionController.cs [hunk 5]
```diff
@@ -37,5 +37,59 @@ public IActionResult MultipleAnswerQuestionForm(string id, int[] answers)
...
+                {
+                    isRight = false;
+                }
+                allAnswers.Add(new MultipleAnswerQuestionOption(answersText[i], isRight));
+            }
+            MultipleAnswerQuestion question = questionRepository.PickById(Id);
+            question.UpdateQuestion(questionText, allAnswers);
+            questionRepository.SaveAllChanges();
+            return View();
+        }
```
Comment (viktartse) [inline src/Teams/Controllers/MultipleAnswerQuestionController.cs:92]: I get an exception when I try to edit a question.
image

**Review by viktartse — CHANGES_REQUESTED** [review]

File: src/Teams/Models/MAQDTOModel.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Teams.Models
+{
+    public class MAQDTOModel  //Вот моя DTO модель
```
[the remark this answers is not in the document]
Comment (IlyaPrikhach) [reply src/Teams/Models/MAQDTOModel.cs:8]: переименовал мою DTO-модель в MultipleAnswerQuestionDTOModel

File: src/Teams/Models/MAQDTOModel.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    {
+        public string questionText { get; set; }
+        public string[] jTextList { get; set; }
+        public bool[] jCheckboxList { get; set; }
```
[the remark this answers is not in the document]
Comment (IlyaPrikhach) [reply src/Teams/Models/MAQDTOModel.cs:12]: Заменил jTextList и jCheckBoxList на TextMassive и CheckboxValueMassive соответственно. Написал "Massive", потому что в скрипте это может быть неочевидно.

File: src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+@*
+    For more information on enabling MVC for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+*@
+@{
+
+}
+
+
+<meta charset="utf-8">
```
[the remark this answers is not in the document]
Comment (IlyaPrikhach) [reply src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml:11]: Я так понимаю что он не нужен? Убрал

File: src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+<label>Введите текст вопроса:</label>
```
[the remark this answers is not in the document]
Comment (IlyaPrikhach) [reply src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml:12]: Теперь это выглядит так <h3>Введите текст вопроса:</h3>

[the remark this answers is not in the document]
Comment (IlyaPrikhach) [reply src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml:12]: заменил label на h3

File: src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+        checkbox.setAttribute('type', 'checkbox');
+        checkbox.setAttribute('class', 'itTrue');
+
+        element.appendChild(inpt);
+        element.appendChild(checkbox);
+        element.appendChild(br);
+
+        hidenText.hidden = false;
+        answersCount++;
+         if(answersCount == 5 ){
```
[the remark this answers is not in the document]
Comment (IlyaPrikhach) [reply src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml:45]: Создал константу MaxAnswersNumber = 5; Именовал в PascalCase

File: src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+}
+
+
+<meta charset="utf-8">
+<label>Введите текст вопроса:</label>
+<br />
+<div id="qText">
+    <input type="text" name="questionText"><br /><br />
+    <p id="hidenText" hidden>Добавьте текст ответа и отметьте правильный ли он</p>
```
[the remark this answers is not in the document]
Comment (IlyaPrikhach) [reply src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml:16]: В браузере вроде интуитивно понятно. Это сообщение появляется, когда нажимаешь на кнопку "Добавить"

File: src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+</div>
+<input type="button" id="buttonForAdd" value="Добавить вариант ответа" onclick="addForm()" />
+<input type="button" value="Отправить" onclick="addAnswers()" />
+
+
+
+<script>
+    let answersCount = 0;
+
+    function addForm() {
```
[the remark this answers is not in the document]
Comment (IlyaPrikhach) [reply src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml:26]: Заменил на AddAnswer()

File: src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+        }
+        else if(isChecked == 0){
+            alert("Хотя бы один вариант ответа должен быть верным");
+        }
+        else{
+            var model = {                       
+            questionText: questionText,
+            jTextList: jTextList,
+            jCheckboxList: jCheckboxList
+        }
```
[the remark this answers is not in the document]
Comment (IlyaPrikhach) [reply src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml:89]: Вроде исправил

File: src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+        $.ajax({
+            type: "POST",
+            url: "@Url.Action("AddMultipleAnswerQuestion", "MultipleAnswerQuestion")",
+            data: JSON.stringify(model),
+            contentType: "application/json; charset=utf-8",
+            dataType: "html",
+            traditional: true,
+            success: function () {
+                alert("Question is added");
+                console.log('Question is added');
```
[the remark this answers is not in the document]
Comment (IlyaPrikhach) [reply src/Teams/Views/MultipleAnswerQuestion/AddMultipleAnswerQuestion.cshtml:101]: Удалил

**Comment by viktartse** [comment]
Have you pushed you changes?
